public_html/core/engine/dispatcher.php

```diff
@@ -286,86 +271,95 @@ public function dispatch($parent_controller = '')
-                    //If main controller come back for new dispatch
-                    return $dispatch->getController().'/'.$dispatch->getMethod();
+        try {
+            if (is_callable([$controller, $this->method])) {
+                /**
+                 * @var $dispatch ADispatcher
+                 */
+                $r = new ReflectionMethod($controller, $this->method);
+                $params = $r->getParameters();
+                if (!$params) {
+                    $args = [];
                 } else {
-                    // Call new dispatch for new controller and exit
-                    //???? need to put limit for recursion to prevent overflow
-                    $dispatch->dispatch();
-
-                    return null;
+                    $args = $this->args;
                 }
-            } else {
-                if ($dispatch == 'completed') {
-                    //Check if we have message completed in controller response.
-                    //If completed. stop further execution.
-                    return 'completed';
-                }
-            }
-            /**
-             * Load layout and process children controllers
-             * @method AController getChildren()
-             */
-            $children = $controller->getChildren();
-
-            ADebug::variable('Processing children of '.$this->controller, $children);
-
-            //Process each child controller
-            foreach ($children as $child) {
-                //Add highest Debug level here with backtrace to review this
-                ADebug::checkpoint($child['controller'].' ( child of '.$this->controller
-                                   .', instance_id: '.$child['instance_id'].' ) dispatch START');
-                //Process each child and create dispatch to call recursive
-                $dispatch = new ADispatcher($child['controller'], ["instance_id" => $child['instance_id']]);
-                $dispatch->dispatch($controller);
-                // Append output of child controller to current controller
-                if (isset($child['position']) && $child['position']) { // made for recognizing few custom_blocks in the same placeholder
-                    $controller->view->assign($child['block_txt_id'].'_'.$child['instance_id'],
-                        $this->response->getOutput());
+
+                $dispatch = call_user_func_array([$controller, $this->method], $args);
+                //Check if return is a dispatch and need to call new page
+                if ($dispatch && is_object($dispatch)) {
+                    if ($this->args["instance_id"] == 0) {
+                        //If main controller come back for new dispatch
+                        return $dispatch->getController().'/'.$dispatch->getMethod();
+                    } else {
+                        // Call new dispatch for new controller and exit
+                        //???? need to put limit for recursion to prevent overflow
```

Comment:
Should we address this issue? If not, we need to remove the comment

Reply:
no sure